llmfoundry/data/finetuning/tasks.py

```diff
@@ -718,7 +718,72 @@ def state_dict(self, num_samples: int,
             num_samples=num_samples,
             from_beginning=from_beginning,
         )
+    
+def maybe_safe_download_hf_data(
+    dataset_name: str,
+    hf_kwargs: Optional[dict[str, Any]] = None
+) -> str:
+    """
+    Download a HuggingFace dataset locally if it does not already exist.
+
+    Args:
+        dataset_name (str): The name of the HuggingFace dataset to use. Can be a remote http(s) 
+        directory or object store bucket containing the file {split}.jsonl.
+        hf_kwargs (dict, optional): Additional kwargs to pass to `datasets.load_dataset`.
+
+    Returns:
+        str: The local path to the dataset.
+    """
+    if hf_kwargs is None:
+        hf_kwargs = {}
+
+    if not os.path.isdir(dataset_name):
+        local_dataset_dir = os.path.join(
+            DOWNLOADED_FT_DATASETS_DIRPATH,
+            dataset_name,
+        )
+
+        if _is_empty_or_nonexistent(dirpath=local_dataset_dir):
+            # Safely load the dataset from HF Hub with restricted file types.
+            hf_hub.snapshot_download(
```

Comment:
A note that I was previously running into cache errors when downloading HF data. The `safe_load` check would fail if the data was not present in cache. I couldn't diagnose the exact cause. 

This isn't blocking if we assume no one is using eval + a HF dataset (which I think is fair).

Reply:
Honestly should prob not support HF dataset splitting + throw an error. Just have all our e2e tests use some UC volume

Reply:
yeah i think we had discussed just not supporting HF dataset for this to simplify things

Reply:
👍 this also simplifies the HF regex issue above

Reply:
removed splitting support for hf datasets